core/src/main/scala/kafka/tools/StreamsResetter.java

```diff
@@ -314,30 +319,47 @@ private boolean isIntermediateTopic(final String topic) {
         return options.valuesOf(intermediateTopicsOption).contains(topic);
     }
 
-    private void maybeDeleteInternalTopics(final ZkUtils zkUtils) {
+    private void maybeDeleteInternalTopics(final KafkaAdminClient adminClient) {
 
         System.out.println("Deleting all internal/auto-created topics for application " + options.valueOf(applicationIdOption));
-
-        for (final String topic : allTopics) {
-            if (isInternalTopic(topic)) {
-                try {
-                    if (!dryRun) {
-                        final TopicCommand.TopicCommandOptions commandOptions = new TopicCommand.TopicCommandOptions(new String[]{
-                            "--zookeeper", options.valueOf(zookeeperOption),
-                            "--delete", "--topic", topic});
-                        TopicCommand.deleteTopic(zkUtils, commandOptions);
-                    } else {
-                        System.out.println("Topic: " + topic);
-                    }
-                } catch (final RuntimeException e) {
-                    System.err.println("ERROR: Deleting topic " + topic + " failed.");
-                    throw e;
+        List<String> topicsToDelete = new ArrayList<>();
+        for (final String listing : allTopics) {
+            if (isInternalTopic(listing)) {
+                if (!dryRun) {
+                    topicsToDelete.add(listing);
+                } else {
+                    System.out.println("Topic: " + listing);
                 }
             }
         }
+        if (!dryRun) {
+            doDelete(topicsToDelete, adminClient);
+        }
         System.out.println("Done.");
     }
 
+    private void doDelete(final List<String> topicsToDelete,
+                          final KafkaAdminClient adminClient) {
+        RuntimeException deleteException = null;
+        final DeleteTopicsResult deleteTopicsResult = adminClient.deleteTopics(topicsToDelete);
+        final Map<String, KafkaFuture<Void>> results = deleteTopicsResult.values();
+
+        for (Map.Entry<String, KafkaFuture<Void>> entry : results.entrySet()) {
+            try {
+                entry.getValue().get(30, TimeUnit.SECONDS);
+            } catch (Exception e) {
+                System.err.println("ERROR deleting topic " + entry.getKey() + " failed " + e.getMessage());
```

Comment:
nit: `Error[:]`

As we do print all errors here, do we need to include the first detected exception when failing? This implies, that we log this exception twice. Maybe we should just print the whole Stacktrace for each exception, and go a generic error message at the end?